telegram/utils/helpers.py

```diff
@@ -40,54 +44,148 @@ def get_signal_name(signum):
+    Any objects from the :module:`datetime` module that are timezone-naive will be assumed
+    to be in UTC.
+
+    ``None`` s are left alone (i.e. ``to_float_timestamp(None)`` is ``None``).
+
     Args:
-        dt_obj (:class:`datetime.datetime`):
+        t (int | float | datetime.timedelta | datetime.datetime | datetime.time | None):
+            Time value to convert. The semantics of this parameter will depend on its type:
+
+            * :obj:`int` or :obj:`float` will be interpreted as "seconds from ``reference_t``"
+            * :obj:`datetime.timedelta` will be interpreted as
+              "time increment from ``reference_t``"
+            * :obj:`datetime.datetime` will be interpreted as an absolute date/time value
+            * :obj:`datetime.time` will be interpreted as a specific time of day
+
+        reference_timestamp (float, optional): POSIX timestamp that indicates the absolute time
+            from which relative calculations are to be performed (e.g. when ``t`` is given as an
+            :obj:`int`, indicating "seconds from ``reference_t``"). Defaults to now (the time at
+            which this function is called).
 
     Returns:
-        int:
+        (float | None) The return value depends on the type of argument ``t``. If ``t`` is
+            given as a time increment (i.e. as a obj:`int`, :obj:`float` or
+            :obj:`datetime.timedelta`), then the return value will be ``reference_t`` + ``t``.
 
+            Else if it is given as an absolute date/time value (i.e. a :obj:`datetime.datetime`
+            object), the equivalent value as a POSIX timestamp will be returned.
+
+            Finally, if it is a time of the day without date (i.e. a :obj:`datetime.time`
+            object), the return value is the nearest future occurrence of that time of day.
+
+            If ``t`` is ``None``, ``None`` is returned (to facilitate formulating HTTP requests
+            when the object to be serialized has a ``date`` which is ``None`` without having to
+            check explicitly).
     """
-    if not dt_obj:
+
+    if reference_timestamp is None:
+        reference_timestamp = time.time()
+
+    if t is None:
         return None
+    elif isinstance(t, dtm.timedelta):
+        return reference_timestamp + t.total_seconds()
+    elif isinstance(t, Number):
+        return reference_timestamp + t
+    elif isinstance(t, dtm.time):
+        reference_dt = dtm.datetime.fromtimestamp(reference_timestamp, tz=t.tzinfo) if t.tzinfo \
+            else dtm.datetime.utcfromtimestamp(reference_timestamp)  # assume UTC for naive
+        reference_date, reference_time = reference_dt.date(), reference_dt.timetz()
+        if reference_time > t:  # if the time of day has passed today, use tomorrow
+            reference_date += dtm.timedelta(days=1)
+        t = dtm.datetime.combine(reference_date, t)
 
-    return int(_timestamp(dt_obj))
+    if isinstance(t, dtm.datetime):
+        return _datetime_to_float_timestamp(t)
```

Comment:
Yes, I should definitely add this

Reply:
Fixed in cc40d63a. I moved that at the top with the `None` check for `reference_timestamp`, so that we don't have to store an `original_reference_timestamp`. Is that okay?